support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -75,24 +78,36 @@ class ZuoraSubscriptionCreator(
 
     val environment = TouchPointEnvironments.fromStage(Configuration.stage, state.user.isTestUser)
 
+    val maybeGiftRecipWithCode = state.giftRecipient.map(maybeAddCode(state.product.billingPeriod, _))
+      .withLogging("gift recipient with code")
```

Comment:
now we generate the code up front rather than in the builder.  This does complicate this top level a bit, but the alternative of changing the type returned by the builder so we could pick out the code, was not turning out to be nice.